[Bug]: `wstring_view` fails to build on LLVM embedded ARM

### Bug Description


When building MP units on LLVM 20.1.0 from [arm-toolchain/releases/20.1.0-ATfE](https://github.com/arm/arm-toolchain/releases/tag/release-20.1.0-ATfE), I get the following error:

```
In file included from /home/runner/.conan2/p/mp-un630ed7150a59e/s/src/core/mp-units-core.cpp:35:
In file included from /home/runner/.conan2/p/mp-un630ed7150a59e/s/src/core/include/mp-units/core.h:27:
In file included from /home/runner/.conan2/p/mp-un630ed7150a59e/s/src/core/include/mp-units/concepts.h:26:
In file included from /home/runner/.conan2/p/mp-un630ed7150a59e/s/src/core/include/mp-units/framework/dimension_concepts.h:28:
In file included from /home/runner/.conan2/p/mp-un630ed7150a59e/s/src/core/include/mp-units/framework/symbol_text.h:31:
/home/runner/.conan2/p/mp-un630ed7150a59e/s/src/core/include/mp-units/ext/fixed_string.h:303:63: error: no member named 'wstring_view' in namespace 'std'
  303 | struct std::hash<mp_units::fixed_wstring<N>> : std::hash<std::wstring_view> {};
      |                                                          ~~~~~^
1 error generated.
```

This is due to LLVM's `wstring_view` definition being gated by the `_LIBCPP_HAS_WIDE_CHARACTERS` macro and which is disabled for ARM embedded target builds:

```
#if _LIBCPP_HAS_WIDE_CHARACTERS
typedef basic_string_view<wchar_t> wstring_view;
#endif
```

Which can be found here: https://github.com/llvm/llvm-project/blob/5dcd3816ed700e792bf2fdaeae7b7001cec821cf/libcxx/include/__fwd/string_view.h#L31-L33

In my fork of mp-units (for my own conan package repo), I get this error on my build with llvm https://github.com/libhal/mp-units/actions/runs/20395734360/job/58610962077

### Steps to Reproduce

Build mp-units using `https://github.com/arm/arm-toolchain/releases/tag/release-20.1.0-ATfE`. 

I have tried setting `_LIBCPP_HAS_WIDE_CHARACTERS=0` on my build for mac I do not get the behavior shown above.


### 🚀 Compiler Explorer Link (Recommended)

_No response_

### Compiler

Clang 20

### C++ Standard

C++20

### mp-units Version

2.5.0

### Build Configuration

- [x] Using Conan
- [ ] Using CMake directly
- [ ] Header-only mode
- [x] C++ modules enabled
- [x] Freestanding mode
- [ ] Debug build
- [ ] Release build

### Additional Context

_No response_

### Checklist

- [x] I have read the documentation and checked existing issues

--- a/src/core/include/mp-units/ext/fixed_string.h
+++ b/src/core/include/mp-units/ext/fixed_string.h
@@ -299,8 +299,13 @@ template<std::size_t N>
 struct std::hash<mp_units::fixed_u16string<N>> : std::hash<std::u16string_view> {};
 template<std::size_t N>
 struct std::hash<mp_units::fixed_u32string<N>> : std::hash<std::u32string_view> {};
+
+// NOTE: We use std::basic_string_view<wchar_t> instead of std::wstring_view because LLVM only makes std::wstring_view
+// available when the macro _LIBCPP_HAS_WIDE_CHARACTERS is set to 1. The `wchar_t` type can continue to be used
+// without that macro definition. To avoid requiring an additional macro, we simply use std::basic_string_view<wchar_t>,
+// which is the underlying definition of std::wstring_view.
 template<std::size_t N>
-struct std::hash<mp_units::fixed_wstring<N>> : std::hash<std::wstring_view> {};
+struct std::hash<mp_units::fixed_wstring<N>> : std::hash<std::basic_string_view<wchar_t>> {};
 
 #if MP_UNITS_HOSTED
 // formatting support

